src/backend/joanie/core/factories.py

```diff
@@ -671,9 +671,12 @@ def main_invoice(self, create, extracted, **kwargs):
                 extracted.save()
                 return extracted
 
-            if self.state == enums.ORDER_STATE_COMPLETED:
-                # If the order is not fee and its state is validated, create
-                # a main invoice with related transaction.
+            if self.state in [
+                enums.ORDER_STATE_TO_SAVE_PAYMENT_METHOD,
+                enums.ORDER_STATE_COMPLETED,
+            ]:
+                # If the order is not free and its state is `completed` or
+                # `to_save_payment_method` create a main invoice with related transaction.
```

Comment:
Why not creating the `main_invoice` for each state except draft like you did for `OrderGeneratorFactory`. This would stick better to the reality.

Reply:
It's possible, and I will apply it :+1:

Reply:
Done ! ;)